keps/sig-node/1972-kubelet-exec-probe-timeouts/README.md

```diff
@@ -0,0 +1,108 @@
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [X] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [X] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [X] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubelet today does not respect exec probe timeouts. This is considered a bug we should fix since
+the timeout value is supported in the Container Probe API. Because exec probe timeouts
+were never respected by kubelet, a new feature gate `ExecProbeTimeouts` will be introduced
+so users have an easy way to revert back if the newly introduced probe timeout results
+in unexpected behavior.
+
+## Motivation
+
+Kubelet not respecting the probe timeout is a bug and should be fixed.
+
+### Goals
+
+* fix exec probe timeouts in kubelet
+
+### Non-Goals
+
+* ensuring exec processes that timed out have been killed by kubelet.
+
+## Proposal
+
+### Risks and Mitigations
+
+* existing workloads on Kubernetes that relied on this bug may unexpectedly see their probes timeout
+
+## Design Details
+
+Changes to kubelet:
+* Ensure kubelet handles timeout errors and registers them as failing probes.
+* Add feature gate `ExecProbeTimeouts` that is GA and on by default.
+* If the feature gate `ExecProbeTimeouts` is disabled and an exec probe timeout is reached, add warning logs to inform users that exec probes are timing out.
+* Re-enable existing exec liveness probe e2e test.
+* Add new exec readiness probe e2e test.
+
```

Comment:
should the new timeout error code introduction be part of this KEP? Or put it into non-goal?

Reply:
Added to non-goals